portal_blocks_documentation/docs.py

```diff
@@ -81,48 +101,145 @@ def write_official_doc(doc_json_file: pathlib.Path, block_name: str):
+        startIndex = -1
+        endIndex = -1
+        entries = MENU_FILE.read_text().split("\n")
         for index, entry in enumerate(entries):
-            if sub_lookup in entry:
-                entries.insert(index, entry.split('#')[0] + menu_entry_ref)
-                entries.insert(index, entry.split('#')[0] + menu_entry_name)
+            if sub_lookup_start in entry:
+                startIndex = index
+            if sub_lookup_end in entry:
+                endIndex = index
+            if startIndex > -1 and endIndex > -1:
                 break
-        MENU_FILE.write_text('\n'.join(entries))
+        if endIndex - startIndex > 1:
+            for i in range(startIndex, endIndex-1):
+                deletedEntry = entries.pop(startIndex + 1)
+                logger.debug(f"deleted entry from menu: {deletedEntry}")
+        else:
+            logger.debug(
+                f"No menu entries to be cleaned up in defined section (start: {startIndex}, end: {endIndex}"
+            )
+        MENU_FILE.write_text("\n".join(entries))
 
-def generate():
 
+def insert_menu_entry(menu_entry_name: str, menu_entry_ref: str, insert_before_lookup: str):
+    if MENU_FILE.exists():
+        entries = MENU_FILE.read_text().split("\n")
+        for index, entry in enumerate(entries):
+            if insert_before_lookup in entry:
+                entries.insert(index, entry.split("#")[0] + menu_entry_ref)
+                entries.insert(index, entry.split("#")[0] + menu_entry_name)
+                break
+        MENU_FILE.write_text("\n".join(entries))
+
+
+def generate_maps_doc():
+    timer_start = timer()
+    logger.info("Building map layout docs...")
+    logger.debug("Read map and mode information...")
+    jsonFile = open(MAPS_JSON_FILE)
+    mapsAndModes = json.load(jsonFile)
+    logger.debug("Successfully read map and mode information.")
+    logger.debug("Generate Maps index...")
+    docLines = [
+        "---\n",
+        "title: Map Layouts\n",
+        "geekdocCollapseSection: true\n"
+        "---\n\n",
+        "{{< toc-tree >}}\n\n",
+    ]
+    docFile = open(f"{MAPS_DOC_FILE_DIR}{MAPS_DOC_FILE}", mode="w+", encoding="utf-8")
+    docFile.writelines(docLines)
+    docFile.close()
+    docLines.clear()
+    logger.debug("Successfully generated maps index.")
+    for map in sorted(mapsAndModes["maps"]):
+        logger.debug(f"Generate doc for {map}...")
+        docLines = ["---\n", f"title: {map}\n", "---\n\n", "{{< toc >}}\n\n"]
+        docLines.append(f"# {map}\n\n")
+        docLines.append(
+            f"![{map} thumbnail]({MAPS_THUMBNAIL_IMGS_DIR_MD}{identify(map)}_thumbnail.{MAPS_THUMBNAIL_FILE_EXT})\n\n"
+        )
+        for modeType in mapsAndModes["modes"]:
+            # docLines.append(f'{{{{< expand label="{modeType.capitalize()} Game Modes">}}}}\n')
+            for mode in sorted(mapsAndModes["modes"][modeType]):
+                docLines.append(f"## {map} - {mode} ({modeType})\n\n")
```

Comment:
again please make use of `.join` function at the end when u have to write to the file.. as that makes sure that all lines get a `\n`... see line 78